manifests/init.pp

```diff
@@ -334,6 +335,7 @@
   validate_legacy(Array, 'validate_array', $facts_to_tags)
   validate_legacy(Boolean, 'validate_bool', $puppet_run_reports)
   validate_legacy(String, 'validate_string', $puppetmaster_user)
+  validate_legacy(String, 'validate_string', $puppet_gem_provider)
```

Comment:
Instead of using legacy stuff, let's use data types for the arguments. We can just remove this if we use `String` type above.